Inferences/TermAlgebraReasoning.cpp

```diff
@@ -248,16 +246,22 @@ namespace Inferences {
     return c;
   }
 
-  bool NegativeInjectivityISE::litCondition(Clause *c, unsigned i) {
+  bool NegativeInjectivityISE::litCondition(Clause *c, unsigned i)
+  {
+    CALL("NegativeInjectivityISE::litCondition");
     Literal *lit = (*c)[i];
     if (sameConstructorsEquality(lit) && !lit->polarity()) {
       unsigned arity = lit->nthArgument(0)->term()->arity();
-      OperatorType *type = env.signature->getFunction(lit->nthArgument(0)->term()->functor())->fnType();
       for (unsigned j = 0; j < arity; j++) {
+        if (j < lit->nthArgument(0)->term()->numTypeArguments()) {
```

Comment:
I think this is not necessary due to the well-typedness which is performed on each shared term. @ibnyusuf what do you think?

Reply:
Agreed. This check is unnecessary, since if the type arguments differed, the equality literal wouldn't type check.